### PR TITLE
GameUI working with gameplay

### PR DESCRIPTION
**Pull request by RSGameDev**
- Connected the scoring with gameplay so when the enemy dies 'on beat' the combo metre increments. If the enemy dies 'off beat' then the combo metre does not increment.
- Once the grade levels-up the multiplier for the score does to.
- Reworked the beatbar so it performs better.
- Removed some errors showing in the console.

i've added a few comments on the code here to explain why i included certain things in the code.

### CODE REVIEW

File: bpdv3/Assets/Scripts/Core/Spawner.cs [hunk 1]
```diff
@@ -90,6 +91,7 @@ private void SpawnObject()
             }
             objectPool.transform.position = SpawnPosition();
             objectPool.SetActive(true);
+            objectPool.GetComponent<Enemy>().hasSpawned = true;
```
Comment (RSGameDev) [inline bpdv3/Assets/Scripts/Core/Spawner.cs:94]: this boolean 'hasSpawned' was created to eliminate an error appearing on the console. Only created for this reason.

File: bpdv3/Assets/Scripts/Floor/TileProperties.cs [hunk 2]
```diff
@@ -20,7 +20,7 @@ public class TileProperties : MonoBehaviour
 
         #region Public & Protected variables
 
-        public int tileWithToken;
+        public int tileWithToken = 0;
```
Comment (RSGameDev) [inline bpdv3/Assets/Scripts/Floor/TileProperties.cs:23]: i added this to remove an error message from the console also, not sure if it was needed though thinking about again but keeping it does make the variable defined which doesn't hurt.

File: bpdv3/Assets/Scripts/UI/Main/BeatBar.cs [hunk 3]
```diff
@@ -42,6 +48,7 @@ void Start()
         // Update is called once per frame
         void Update()
         {
+            threshHoldValue = thresholdZone;
```
Comment (RSGameDev) [inline bpdv3/Assets/Scripts/UI/Main/BeatBar.cs:51]: oops, i thought i removed this. I need to remove this. I made just so i could see the static variables value in the inspector.

File: bpdv3/Assets/Scripts/enemy/Enemy.cs [hunk 4]
```diff
@@ -38,8 +42,12 @@ public enum EnemyType
 
         private void OnDisable()
         {
-            token = true;
+            if (hasSpawned)
```
Comment (RSGameDev) [inline bpdv3/Assets/Scripts/enemy/Enemy.cs:45]: this was the main reason for making the hasSpawned variable. Including this removes the error message that appeared in the console. You will see similar on the enemynextmove script also.

File: bpdv3/Assets/Scripts/enemy/EnemyNextMove.cs [hunk 5]
```diff
@@ -29,7 +31,10 @@ public class EnemyNextMove : MonoBehaviour
 
         private void OnDisable()
         {
-            NewCycle();
+            if (_enemy.hasSpawned)
```
Comment (RSGameDev) [inline bpdv3/Assets/Scripts/enemy/EnemyNextMove.cs:34]: added to remove error messages appearing on the console.

File: bpdv3/Assets/Scripts/Floor/TileProperties.cs (hunk 2 above)
Comment (tikokito123) [reply bpdv3/Assets/Scripts/Floor/TileProperties.cs:23]: That is why you initialize it to null

**Review by tikokito123 — COMMENTED** [review]
...

**Pull request merged**